3rdParty/StormLib/src/FileStream.cpp

```diff
@@ -21,6 +21,14 @@
 #ifdef _MSC_VER
 #pragma comment(lib, "wininet.lib")             // Internet functions for HTTP stream
 #pragma warning(disable: 4800)                  // 'BOOL' : forcing value to bool 'true' or 'false' (performance warning)
+#endif
+
+#ifdef __HAIKU__
```

Comment:
It looks like stormlib solved the same issue for FreeBSD by settings this in CMake:
https://github.com/ladislav-zezula/StormLib/blob/master/CMakeLists.txt#L303

I think we should go for the same solution here instead of modifying the stormlib source localy.

Reply:
O_LARGEFILE should be disabled in StormPort.h
https://github.com/ladislav-zezula/StormLib/blob/4ad0bff21da0163917e1552960e6a43679586b89/src/StormPort.h#L202

It looks like this is also the place to define *64 functions, not sure why they ended up putting it in CMake for FreeBSD :/